### PR TITLE
feat(crashlytics): Firbase web modular V9 API

### PR DESCRIPTION
**Pull request by exaby73**
### Checklist

- I read the [Contributor Guide](../CONTRIBUTING.md) and followed the process outlined there for submitting PRs.
  - [ ] Yes
- My change supports the following platforms;
  - [ ] `Android`
  - [ ] `iOS`
- My change includes tests;
  - [ ] `e2e` tests added or updated in `packages/\*\*/e2e`
  - [ ] `jest` tests added or updated in `packages/\*\*/__tests__`
- [ ] I have updated TypeScript types that are affected by my change.
- This is a breaking change;
  - [ ] Yes
  - [ ] No

### Test Plan

---

Think `react-native-firebase` is great? Please consider supporting the project with any of the below:

- 👉 Star this repo on GitHub ⭐️
- 👉 Follow [`React Native Firebase`](https://twitter.com/rnfirebase) and [`Invertase`](https://twitter.com/invertaseio) on Twitter

### CODE REVIEW

**Review by russellwheatley — CHANGES_REQUESTED** [review]
Nice work. Much better having no duplicate tests running. Like that we have a declaration file for the modular API. Only one change from me 👍 

iOS CI seems flaky and unrelated. Also - code coverage CI is picking up code that is covered, not sure what to do about that but ingnore?

File: packages/crashlytics/lib/modular/index.js [hunk 1]
```diff
@@ -0,0 +1,116 @@
+import { firebase } from '..';
+
+/**
+ * @param {import('@react-native-firebase/app').ReactNativeFirebase.Module} app
+ * @returns {import('.').FirebaseCrashlytics}
+ */
+export function getCrashlytics(app) {
```
[the remark this answers is not in the document]
Comment (exaby73) [reply packages/crashlytics/lib/modular/index.js:7]: Fixed

**Comment by exaby73** [comment]
Code coverage has been having trouble with the modular APIs. Most if not all of the code is tested, but some functions that can't be tested (and that weren't tested in the V8 APIs either) cause the coverage to go down. Not sure what to do about those. I don't feel writing noop tests to make coverage happy is a good idea so I left it so.

As for the flakiness, it has always been flaky but now since we are essentially testing the same flaky code twice, just wrapped in modular functions, the probability of flakiness showing itself is higher.

**Review by russellwheatley — DISMISSED** [review]

**Comment by mikehardy** [comment]
Looking through this now - as a general comment though - please don't mix feature/fix changes with dependency changes, and please don't do merge commits. Much better to rebase off main and repush. I'll do so and drop the various deps + merge commit stuff so the PR is easier to review

CI should not be flaky either - that's something that should be a specific comment as needed. It *does* flake but I try to squash the flakes as soon as possible - that was actually a focus end of last week and I think things are purring along pretty well now

File: packages/crashlytics/lib/modular/index.js [hunk 2]
```diff
@@ -0,0 +1,116 @@
+import { firebase } from '..';
+
+/**
+ * @param {import('@react-native-firebase/app').ReactNativeFirebase.Module} app
```
Comment (mikehardy) [inline packages/crashlytics/lib/modular/index.js:4]: I don't understand the typing here in the params/returns in the docs?

File: tests/e2e/.mocharc.js [hunk 3]
```diff
@@ -10,22 +10,22 @@ module.exports = {
   exit: true,
   require: 'node_modules/jet/platform/node',
   spec: [
-    '../packages/app/e2e/**/*.e2e.js',
-    '../packages/app-check/e2e/**/*.e2e.js',
-    '../packages/app-distribution/e2e/**/*.e2e.js',
-    '../packages/analytics/e2e/**/*.e2e.js',
-    '../packages/auth/e2e/**/*.e2e.js',
+    // '../packages/app/e2e/**/*.e2e.js',
+    // '../packages/app-check/e2e/**/*.e2e.js',
```
Comment (mikehardy) [inline tests/e2e/.mocharc.js]: I see these changes were reverted in a following commit - can just rebase them out of the commit chain completely with a `rebase -i` (probably `rebase -i HEAD~1` then mark the revert as a "squash", edit the commit notes in the next screen and poof, they weren't there to begin with

File: packages/crashlytics/lib/modular/index.js (hunk 2 above)
Comment (mikehardy) [reply packages/crashlytics/lib/modular/index.js]: fixed using 

```typescript
/**
 * @typedef {import("..").FirebaseApp} FirebaseApp
 * @typedef {import("..").FirebaseDatabaseTypes.Module} FirebaseCrashlytics
 */
```

at top of file per suggestion by @russellwheatley

File: packages/crashlytics/lib/index.d.ts [hunk 4]
```diff
@@ -77,7 +77,7 @@ export namespace FirebaseCrashlyticsTypes {
      * ```
      *
      */
-    isCrashlyticsCollectionEnabled: true;
+    isCrashlyticsCollectionEnabled: boolean;
```
Comment (mikehardy) [inline packages/crashlytics/lib/index.d.ts:80]: nice catch!

**Review by mikehardy — APPROVED** [review]
still investigating the test coverage notes saying there is no coverage when it *appears* there should be but types are fixed with the typedef suggestion, and I pushed that.

If e2e CI run gets the coverage I expect this will merge immediately, otherwise I'll keep investigating

**Comment by mikehardy** [comment]
Crashlytics modular methods are not being covered in CI coverage reports

Need to investigate why not, but don't want to block the PR so this is a separate issue

**Pull request merged**